src/autopas/containers/verletClusterLists/VerletClusterListsRebuilder.h

```diff
@@ -414,24 +418,36 @@ class VerletClusterListsRebuilder {
    * @return Tower reference.
    */
   auto &getTower(std::array<double, 3> location) {
-    std::array<size_t, 2> towerIndex{};
+    auto [towerIndexX, towerIndexY] = getTowerCoordinates(location);
+    return getTower(towerIndexX, towerIndexY);
+  }
+
+  /**
+   * Returns the coordinates of the tower in the tower grid the given 3D coordinates are in.
+   * If the location is outside of the domain, the tower nearest tower is returned.
+   *
+   * @param location The 3D coordinates.
+   * @return Tower reference.
+   */
+  std::array<size_t, 2> getTowerCoordinates(std::array<double, 3> location) {
+    std::array<size_t, 2> towerIndex2D{};
 
     for (int dim = 0; dim < 2; dim++) {
       const auto towerDimIndex =
-          (static_cast<long int>(floor((location[dim] - _haloBoxMin[dim]) * _towerSideLengthReciprocal))) + 1l;
+          (static_cast<long int>(floor((location[dim] - _haloBoxMin[dim]) * _towerSideLengthReciprocal)));
```

Comment:
```suggestion
          static_cast<long int>(floor((location[dim] - _haloBoxMin[dim]) * _towerSideLengthReciprocal));
```

Reply:
there might also be problems with rounding when choosing sth. directly at a boundary of a cell.

Reply:
seems to be fine in the unit tests that should check this